linkerd/app/src/env.rs

```diff
@@ -998,6 +1063,37 @@ mod tests {
         assert_eq!(parse_duration("1"), Err(ParseError::NotADuration));
     }
 
+    #[test]
+    fn convert_labels_to_map_different_values() {
+        let labels_string = "cluster=\"test-cluster1\"\nrack=\"rack-22\nzone=us-est-coast\nlinkerd.io/control-plane-component=\"controller\"\nlinkerd.io/proxy-deployment=\"linkerd-controller\"\nworkload=\nkind=\"\"\nkey1=\"=\"\nkey2==value2\nkey3\n=key4\n
+        "
+        .to_string();
+
+        let expected = [
+            ("cluster".to_string(), "test-cluster1".to_string()),
+            ("rack".to_string(), "rack-22".to_string()),
+            ("zone".to_string(), "us-est-coast".to_string()),
+            (
+                "linkerd.io/control-plane-component".to_string(),
+                "controller".to_string(),
+            ),
+            (
+                "linkerd.io/proxy-deployment".to_string(),
+                "linkerd-controller".to_string(),
+            ),
+            ("workload".to_string(), "".to_string()),
+            ("kind".to_string(), "".to_string()),
+            ("key1".to_string(), "=".to_string()),
+            ("key2".to_string(), "=value2".to_string()),
+            ("".to_string(), "key4".to_string()),
+        ]
+        .iter()
+        .cloned()
```

Comment:
I think we could remove the need to clone here if we changed this to 
```suggestion
        .into_iter()
```
which moves the values out of the array by value, instead of iterating over references.

Reply:
Hmm, But we need a  `std::iter::Iterator<Item=(std::string::String, std::string::String)>`
So the compiler complains as

`value of type "std::collections::HashMap<std::string::String, std::string::String>" cannot be built from "std::iter::Iterator<Item=&(std::string::String, std::string::String)>"`

Reply:
ah, never mind. not a big deal, anyway.